pkg/registry/template/template.go

```diff
@@ -0,0 +1,130 @@
+package template
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+
+	"github.com/golang/glog"
+	// TODO: Fix this import
+	"k8s.io/kubernetes/pkg/util/labels"
+	"k8s.io/kubernetes/pkg/util/stringreplace"
+)
+
+var parameterExp = regexp.MustCompile(`\$\{([a-zA-Z0-9\_]+)\}`)
```

Comment:
will need to change {} to () and in general the substitution logic will need to be updated to also find $(()) and handle the quoting separately.